SConstruct

```diff
@@ -430,6 +437,11 @@ def scons():
             Exit('stdatomic.h is required to compile DAOS, update your compiler or distro version')
         config.Finish()
 
+    scons_ver = get_scons_version(deps_env)
+    if scons_ver >= (4, 0, 0):
+        deps_env.Tool("compilation_db")
+        deps_env.Alias("compiledb", deps_env.CompilationDatabase())  # pylint: disable=no-member
```

Comment:
The way we normally handle these is to add a fake function in fake_scons

https://github.com/daos-stack/daos/tree/master/utils/sl/fake_scons/SCons